gzip_test.go

```diff
@@ -325,17 +325,35 @@ func TestFlushBeforeWrite(t *testing.T) {
 }
 
 func TestImplementCloseNotifier(t *testing.T) {
+	request := &http.Request{}
+	request.Header = http.Header{}
```

Comment:
Mind using https://golang.org/pkg/net/http/httptest/#NewRequest here instead of building the request and header in 2 lines?